### PR TITLE
allow for disabling IPv6 in openhabian.conf

### LINKED ISSUE #228: openhabian-config tries ipv6 and fails
Problem:
openhabian-config tries to install like mosquitto over ipv6 and times out

I forced my apt to use -o Acquire::ForceIPv4=true but that does not work with openhabian-config
An option to select this would be good.

### LINKED ISSUE #887: disable IPv6 ?
See https://community.openhab.org/t/openhabian-setup-fails-due-to-ipv6/98717 and elsewhere on the net for people to have a problem with Raspbian to use IPv6 (only) on boot.

See https://raspberrypi.stackexchange.com/questions/23884/force-ipv4-address for a potential way to change/fix that in openHABian.

Opinions ?

### PR DESCRIPTION
**Pull request by mstormi**
There's reports of openHABian failing to start or hang in installation in some environments to use a partial (but mostly unknown) IPv6 setup.
This PR now allows for complete disablement of IPv6 even prior to first boot by adding `ipv6=disable` in `openhabian.conf`

Fixes #228 
Fixes #887 

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: build-image/openhabian.pi-raspbian.conf [hunk 1]
```diff
@@ -28,12 +28,13 @@ wifi_country=""
...
 # debug verbosity mode
-# unattended, unattended_debug (verbose output in log) or debug_maximum (show every command)
+# values: "unattended", "unattended_debug" (verbose output in log) or "debug_maximum" (show every command in either mode)
 mode=unattended
 
 # Java architecture mode
-# Use 32-bit or 64-bit
-java_arch=32-bit
+# values: "Zulu32", "Zulu64" or "AdoptOpenJDK"
+java_arch=64-bit
```
Comment (ecdye) [inline build-image/openhabian.pi-raspbian.conf:38]: Why switch to 64-bit default on RPi?

File: docs/openhabian-DEBUG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ Next, you need your router (or a different device) to provide properly configure
 The DHCP server also has to announce which DNS resolver to use so your box knows how to translate DNS names into IP addresses.
 It also needs to announce which IP address to use as the default gateway to the internet - a typical access router is also the DHCP server will announce it's own address here.
 Finally, the DHCP server should also announce the NTP server(s) to use for proper time services. Lack thereof will not break the installation procedure but can lead to all sorts of long term issues so we recommend to setup DHCP to announce a reachable and working NTP server.
+A note on IPv6: openHABian was reported failing to boot in some environments that make use of IPv6. If basic IP initialization fails (you cannot `ping` your box) or  installation gets stuck trying to download software packages, you might want to disabling IPv6. You can also do that before the very first install attempt if you're sure you don't need any IPv6 connectivity on your openHABian box. See [this section of openhabian.md](openhabian.md#ipv6-notes) how to disable IPv6 on your system.
```
Comment (ecdye) [inline docs/openhabian-DEBUG.md:29]: "you might want to disabling IPv6." should be "you might want to try disabling IPv6."

File: docs/openhabian.md [hunk 3]
```diff
@@ -139,11 +139,19 @@ For the setup on Wi-Fi, you'll need to make your SSID and password known to the
...
-- Uncomment and fill in `wifi_ssid="My Wi-Fi SSID"` and `wifi_psk="password123"`
+- Uncomment and complete the lines reading `wifi_ssid="My Wi-Fi SSID"` and `wifi_psk="password123"`
 - Save, Unmount, Insert, Boot
-- Continue with the instructions for the Raspberry Pi or Pine A64
+- Continue with the instructions for your hardware
+
+{: #ipv6-notes}
+### IPv6 notes
+
+You might encounter problems when you make use of IPv6 of some sort. openHABian installation may stop or hang forever.
```
Comment (ecdye) [inline docs/openhabian.md:152]: "You might encounter problems when you make use of IPv6 of some sort." could probably be better said: "You might encounter problems when you make use of IPv6 on some networks and systems."

File: docs/openhabian.md [hunk 4]
```diff
@@ -279,16 +288,20 @@ During and after the first boot of your Raspberry Pi, the green on-board LED wil
...
-- Raspberry Pi image setup: [http://openhab:8080](http://openhab:8080)
-- In any case: [http://your-device-hostname:8080](http://your-device-hostname:8080) or [http://192.168.0.2:8080](http://192.168.0.2:8080) (replace name/IP)
+Double-check the address and name with your router while you wait.
+
+If there is absolutely no output for more than 10 minutes, your installation has failed in the first initialization phase. There probably is a problem
+with the way your router or local network are setup.
 
-##### SSH Progress Report
+You might want to try disabling IPv6.
+Read on in the [Troubleshooting] section or move on to the [DEBUG guide](docs
```
Comment (ecdye) [inline docs/openhabian.md:301]: Incomplete link to Debug guide

File: build-image/openhabian.pi-raspbian.conf (hunk 1 above)
Comment (mstormi) [reply build-image/openhabian.pi-raspbian.conf]: Ouch, good catch. That was a copy 'n paste mistake.

File: docs/openhabian-DEBUG.md (hunk 2 above)
Comment (mstormi) [reply docs/openhabian-DEBUG.md]: done

File: docs/openhabian.md (hunk 3 above)
Comment (mstormi) [reply docs/openhabian.md]: done

File: docs/openhabian.md (hunk 4 above)
Comment (mstormi) [reply docs/openhabian.md]: done

**Comment by ecdye** [comment]
LGTM, may not be the best solution for long term as outlined by @gitMiguel, but as a temporary fix it looks great. Overall, the ability to disable IPv6 is a nice to have either way.

**Comment by mstormi** [comment]
> @mstormi Perhaps this should be added as a function that can be exposed to openhabian-setup?

what do you mean it IS a function

**Comment by mstormi** [comment]
FTR, it was not intentional to have ipv6=disable per default but to keep it enabled.
So I followed up with #891  to make ipv6=enable the default

**Pull request merged**